src/toil/provisioners/abstractProvisioner.py

```diff
@@ -184,6 +187,25 @@ def readClusterSettings(self):
         """
         raise NotImplementedError
 
+    def _writeGlobalFile(self, key: str, contents: bytes) -> str:
+        """
+        Write a file to a physical storage system that is accessible to the
+        leader and all nodes during the life of the cluster.
+
+        Not to be confused with the `writeGlobalFile()` method in the file
+        store, this is an internal system to store files for the cluster.
+        Additional resources should be cleaned up in `self.destroyCluster()`.
+
+        :return: A public URL that can be used to retrieve the file.
+        """
+        raise NotImplementedError
+
+    def _readGlobalFile(self, key: str) -> bytes:
+        """
+        Return the contents of the file written by `self._writeGlobalFile()`.
```

Comment:
Is this something we ever need to do? Or is it only here for testing?

Reply:
Yeah this is only for testing now, since we don't need this for Ignition.